### ISSUE TITLE
rules engine "add new" starts with connection logic of previously created rule

### ISSUE THREAD

**Issue by kgiori** [mozilla-iot/gateway#1164]
There are a number of issues I'm seeing with the rules engine GUI, but the starting point of the problem could be related to the carry over of the logic from a previous rule. The last rule created by someone else had "while...and...then..." logic. I clicked (+) and saw this image as the starting point:
<img width="1176" alt="screen shot 2018-07-05 at 9 11 52 am" src="https://user-images.githubusercontent.com/9487286/42335107-1d02a722-8034-11e8-9a75-803c8ad00bcf.png">

This had also happened with a straight line logic rule, and when I started dragging and dropping to add elements to the new rule, the line disappeared and there was no obvious way to get it back.

**Comment by rzr** [comment]
Yes I also observed that you can't create a new rule unless you drag back elements to tray, then you can save new rule and observe that an invalid rule was also created.

I can detail or bisect more if needed.

**Comment by rzr** [comment]
Thanks @hobinjk it could have been fixed with:

https://github.com/mozilla-iot/gateway/commit/4228fbccf4794c35ce7398e78570f92d67707b6a

I will verify that soon , and eventually close this ticket.
